src/reducers/cards.js

```diff
@@ -51,6 +31,52 @@ export const cardsReducer = (state = initialState, action) => {
           value: action.value
         }
       }
+    case cardActions.SUBSCRIBE:
+      return {
+        ...state,
+        [action.observableId]: {
+          ...state[action.observableId],
+          ui: {
+            ...state[action.observableId].ui,
+            subscribers: state[action.observableId].ui.subscribers.concat(
+              action.subscriberId
+            )
+          }
+        },
+        [action.subscriberId]: {
+          ...state[action.subscriberId],
+          ui: {
+            ...state[action.subscriberId].ui,
+            childLabels: state[action.subscriberId].ui.childLabels.concat({
+              id: action.observableId,
+              label: state[action.observableId].value
+            })
+          }
+        }
+      }
+    case cardActions.UPDATE_CHILD_LABELS:
+      const result = {}
+
+      Object.keys(state).forEach(
+        (cardId) =>
+          (result[cardId] = state[action.parentId].ui.subscribers.includes(
```

Comment:
we can use some() instead of includes()? but there is still a potential slow down here since we have a double loop

Reply:
Got it down to a single loop that doesn't use `some` or `includes` at all.